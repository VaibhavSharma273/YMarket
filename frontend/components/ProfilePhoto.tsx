import { useState, useEffect } from 'react'
import { Image, StyleSheet } from 'react-native'

export default function ProfilePhoto({ src } : { src: string }): JSX.Element {
  const defaultAvatar = require('../assets/images/blank-profile-picture.png');
  const [image, setImage] = useState(src ? { uri: src} : defaultAvatar);
  
  useEffect(() => {
    if (src != image.uri) {
      setImage(src ? { uri: src} : defaultAvatar);
    }
  }, [src]);

  return <Image source={image} style={styles.image} onError={({ nativeEvent: { error }}) => {
    setImage(defaultAvatar); }}
    />
}

const styles = StyleSheet.create({
  image: {
    width: 100,
    height: 100,
    marginBottom: 20,
    borderRadius: 400/2,
  },
})