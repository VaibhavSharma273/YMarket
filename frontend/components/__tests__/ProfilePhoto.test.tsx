import * as React from 'react';
import renderer from 'react-test-renderer';

import ProfilePhoto from '../ProfilePhoto';

it(`renders correctly`, () => {
  const tree = renderer.create(<ProfilePhoto/>).toJSON();

  expect(tree).toMatchSnapshot();
});
