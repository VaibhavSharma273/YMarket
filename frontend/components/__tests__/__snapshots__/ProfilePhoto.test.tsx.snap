// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  onError={[Function]}
  source={1}
  style={
    Object {
      "borderRadius": 200,
      "height": 100,
      "marginBottom": 20,
      "width": 100,
    }
  }
/>
`;
